drivers/i2c/i2c_ll_stm32_v1.c

```diff
@@ -218,92 +550,74 @@ s32_t stm32_i2c_msg_write(struct device *dev, struct i2c_msg *msg,
 	const struct i2c_stm32_config *cfg = DEV_CFG(dev);
 	struct i2c_stm32_data *data = DEV_DATA(dev);
 	I2C_TypeDef *i2c = cfg->i2c;
-	s32_t ret = 0;
 
-	ARG_UNUSED(next_msg_flags);
+	msg_init(dev, msg, next_msg_flags, saddr, I2C_REQUEST_WRITE);
 
-	data->current.len = msg->len;
-	data->current.buf = msg->buf;
-	data->current.flags = msg->flags;
-	data->current.is_restart = 0U;
-	data->current.is_write = 1U;
-	data->current.is_nack = 0U;
-	data->current.is_err = 0U;
-	data->slave_address = saddr;
-
-	LL_I2C_EnableIT_EVT(i2c);
-	LL_I2C_EnableIT_ERR(i2c);
-	LL_I2C_AcknowledgeNextData(i2c, LL_I2C_ACK);
-	if (msg->flags & I2C_MSG_RESTART) {
-		LL_I2C_GenerateStartCondition(i2c);
-	}
-	LL_I2C_EnableIT_BUF(i2c);
+	stm32_i2c_enable_transfer_interrupts(dev);
+	LL_I2C_EnableIT_TX(i2c);
 
 	k_sem_take(&data->device_sync_sem, K_FOREVER);
 
-	LL_I2C_DisableIT_BUF(i2c);
-	if (data->current.is_nack || data->current.is_err) {
+	return msg_end(dev, next_msg_flags, __func__);
+}
 
-		if (data->current.is_nack)
-			LOG_DBG("%s: NACK", __func__);
+s32_t stm32_i2c_msg_read(struct device *dev, struct i2c_msg *msg,
+			 u8_t *next_msg_flags, u16_t saddr)
+{
+	const struct i2c_stm32_config *cfg = DEV_CFG(dev);
+	struct i2c_stm32_data *data = DEV_DATA(dev);
+	I2C_TypeDef *i2c = cfg->i2c;
 
-		if (data->current.is_err)
-			LOG_DBG("%s: ERR %d", __func__,
-				    data->current.is_err);
+	msg_init(dev, msg, next_msg_flags, saddr, I2C_REQUEST_READ);
 
-		data->current.is_nack = 0U;
-		data->current.is_err = 0U;
-		ret = -EIO;
-	}
+	stm32_i2c_enable_transfer_interrupts(dev);
+	LL_I2C_EnableIT_RX(i2c);
 
-	LL_I2C_DisableIT_EVT(i2c);
-	LL_I2C_DisableIT_ERR(i2c);
+	k_sem_take(&data->device_sync_sem, K_FOREVER);
 
-	return ret;
+	return msg_end(dev, next_msg_flags, __func__);
 }
 
-s32_t stm32_i2c_msg_read(struct device *dev, struct i2c_msg *msg,
-			 u8_t *next_msg_flags, u16_t saddr)
+#else
+
+static inline int check_errors(struct device *dev, const char *funcname)
 {
 	const struct i2c_stm32_config *cfg = DEV_CFG(dev);
 	struct i2c_stm32_data *data = DEV_DATA(dev);
 	I2C_TypeDef *i2c = cfg->i2c;
-	s32_t ret = 0;
-
-	ARG_UNUSED(next_msg_flags);
-
-	data->current.len = msg->len;
-	data->current.buf = msg->buf;
-	data->current.flags = msg->flags;
-	data->current.is_restart = 0U;
-	data->current.is_write = 0U;
-	data->current.is_err = 0U;
-	data->slave_address = saddr;
 
-	LL_I2C_EnableIT_EVT(i2c);
-	LL_I2C_EnableIT_ERR(i2c);
-	LL_I2C_AcknowledgeNextData(i2c, LL_I2C_ACK);
-	LL_I2C_GenerateStartCondition(i2c);
-	LL_I2C_EnableIT_BUF(i2c);
+	if (LL_I2C_IsActiveFlag_AF(i2c)) {
+		LL_I2C_ClearFlag_AF(i2c);
+		data->current.is_nack = 1U;
+		goto error;
+	}
 
-	k_sem_take(&data->device_sync_sem, K_FOREVER);
+	if (LL_I2C_IsActiveFlag_ARLO(i2c)) {
+		LL_I2C_ClearFlag_ARLO(i2c);
+		data->current.is_arlo = 1U;
```

Comment:
To be consistent, please add dbg log message here

Reply:
done